[RFC] Docs cleanup: Mark 'macatsui' option as removed, cleanup todo.txt.

yagebu opened the pull request:
- The obsolete `macatsui` option, which was only relevant for the VIM GUI on Mac, has already been removed in the source, so I have removed it in the documentation too (and mentioned it in `vim_diff.txt`)
- I have removed most GUI-related bugs/comments/todos from `todo.txt`.

This is my first PR for neovim, so I stuck to simple changes and it shouldn't be hard to review. But if potential reviewers want to see smaller or bigger doc-related PRs, let me know!

(refs #2599)

yagebu committed:
doc: mark `macatsui` option as removed, update vim_diff.txt

This option has already been removed in the source.

Nvim does not have a GUI, so `nvim -g` does not work.
Also add `macatsui` to the list of removed options.

yagebu committed:
doc: remove GUI-related comments/bugs from todo.txt

Pyrohh commented:
Thanks for doing this, I'll be sure to look at it later today. I'm not sure what @glts thinks, but it might be a waste of effort to to not touch `todo.txt`. Also just some advice:
- the `docs:` prefix should probably be `doc:`, as it's short for the already plural "documentation"
- the vim_diff.txt update should probably be merged with the first commit.

yagebu commented:
Thanks for the input, will rebase now.

Pyrohh commented on runtime/doc/options.txt:
```diff
@@ -3174,7 +3174,6 @@ A jump table for the options with a short description can be found at |Q_op|.
 <
 	For Mac OSX you can use something like this: >
 	    :set guifont=Monaco:h10
-<	Also see 'macatsui', it can help fix display problems.
```
The `<` at the beginning of the line should be kept. If you're not already, make sure to edit the docs using `conceallevel=0`. I use this mapping, although there's probably many ways to do this:
```
nnoremap <silent> coc
      \ :set conceallevel=<C-r>=&conceallevel == 2 ? 0 : 2<CR><CR>
      \ :set conceallevel?<CR>

```

Pyrohh commented:
> Regarding todo.txt: Most of it is obsolete, but there might be some bugs/comments in there that we should turn into GH issues

Sounds good to me.
By the way, there's a few results for `gtk` and `gui` still in todo.txt, any reason you didn't delete them? I'm guessing you only deleted the things you're sure aren't relevant anymore, not sure though.

Pyrohh commented:
Besides that, this LGTM.

justinmk commented on runtime/doc/todo.txt:
```diff
@@ -2786,9 +2473,6 @@ Most interesting new features to be added when all bugs have been fixed:
     first line to be able to start halfway through.
--   Running a shell command from the GUI still has limitations.  Look into how
-    the terminal emulator of the Vim shell project can help:
-    http://vimshell.wana.at
```
Hey, everyone: this was on Vim's todo list.

yagebu commented:
> I'm guessing you only deleted the things you're sure aren't relevant anymore

This exactly. But I think I'll have another go at it soon and clean it up some more.

Rebased and fixed the help file syntax. Thanks for the mapping!

Pyrohh commented:
Thanks @yagebu, looking forward to more PRs :)

glts commented on runtime/doc/options.txt:
```diff
@@ -4320,20 +4319,7 @@ A jump table for the options with a short description can be found at |Q_op|.
-		endif
-<	Another option to check if you have drawing problems is
-	'termencoding'.
+'macatsui'		Removed. |vim-differences| {Nvim}
```
Why would we want to keep a stub of this legacy option around? It's the first time I see this being done.

Pyrohh replied:
IIRC it's been done for at least a few months, even before `vim-diff.txt`, e.g., 2bf085049718a163ddc71b8c02989158ca595596 and 497a688ebc115c6535360f08464309ede9fba0cf.

> Why would we want to keep a stub of this legacy option around?

So users know that we haven't accidentally removed it (which we've done with some commands), and because a stub saying `Removed.` is arguably more helpful than `Sorry, no help  for salkdjsakldsa`. They take up very little room, so I'd rather we keep them for the practical benefit, at least for the time being.

What do you think?

glts commented:
Ok, I don't mind keeping the note then.

I was thinking that an option in Vim specifically for the legacy Mac GUI was simply not applicable to Neovim at all. Recently we've made `cpo+=H` disappear without a trace because it was useless. But obv I haven't been around for very long so my instinct might be off.

Pyrohh commented:
> I was thinking that an option in Vim specifically for the legacy Mac GUI was simply not applicable to Neovim at all

Perhaps, but I think we should at least document it. I think we can remove them from `options.txt` some time after the first release, while moving their tag definitions to `vim_diff.txt`.
